device/celestica/x86_64-cel_seastone-r0/plugins/sfputil.py

```diff
@@ -109,7 +111,95 @@ def get_low_power_mode(self, port_num):
         return True
 
     def set_low_power_mode(self, port_num, lpmode):
-        raise NotImplementedError
+        # Check for invalid port_num
+        if port_num < self.port_start or port_num > self.port_end:
+            return False
+
+        try:
+            reg_file = open(
+                "/sys/devices/platform/dx010_cpld/qsfp_lpmode", "r+")
+        except IOError as e:
+            print "Error: unable to open file: %s" % str(e)
+            return False
+
+        content = reg_file.readline().rstrip()
+
+        # content is a string containing the hex representation of the register
+        reg_value = int(content, 16)
+
+        # Determind if port_num start from 1 or 0
+        if self.port_start == 1:
+            bit_index = port_num - 1
+        else:
+            bit_index = port_num
+
+        # Mask off the bit corresponding to our port
+        mask = (1 << bit_index)
+
+        # LPMode is active high; set or clear the bit accordingly
+        if lpmode is True:
+            reg_value = reg_value | mask
+        else:
+            reg_value = reg_value & ~mask
+
+        # Convert our register value back to a hex string and write back
+        content = hex(reg_value).strip('L')
+
+        reg_file.seek(0)
+        reg_file.write(content)
+        reg_file.close()
+
+        return True
 
     def reset(self, port_num):
-        raise NotImplementedError
+        # Check for invalid port_num
+        if port_num < self.port_start or port_num > self.port_end:
+            return False
+
+        try:
+            reg_file = open(
+                "/sys/devices/platform/dx010_cpld/qsfp_reset", "r+")
```

Comment:
Please merge above 2 lines into one.

Reply:
Fixed in 5db212356dfce548f532d8398de43fed8dccf973